### ISSUE TITLE
Conda environment for queue jobs

### ISSUE THREAD

**Issue by raimis** [Acellera/htmd#525]
When *Psi4* jobs are queue via Slurm (or the other queues, except the local), *conda* environment activation is lost.

This is problem, if you want to switch to a development version of *Psi4* on a specific *conda* environment.

We need a more general solution to this hack:
```diff
diff --git a/htmd/queues/slurmqueue.py b/htmd/queues/slurmqueue.py
index 009dcb5..71d9f64 100644
--- a/htmd/queues/slurmqueue.py
+++ b/htmd/queues/slurmqueue.py
@@ -141,6 +141,9 @@ class SlurmQueue(SimQueue, ProtocolInterface):
             # Trap kill signals to create sentinel file
             f.write('\ntrap "touch {}" EXIT SIGTERM\n'.format(os.path.normpath(os.path.join(workdir, self._sentinel))))
             f.write('\ncd {}\n'.format(workdir))
+
+            f.write('. /shared/raimis/opt/miniconda3/bin/activate htmd_opt\n')
+
             f.write('{}'.format(runsh))
 
             # Move completed trajectories
```

**Comment by j3mdamas** [comment]
Is this really needed? I think this is a developers features rather than something we want for users. No?

**Comment by j3mdamas** [comment]
OK, this can be implemented in a way similar to what is done on LSF: https://github.com/Acellera/htmd/blob/master/htmd/queues/lsfqueue.py#L148-L150, which allows for arbitrary lines of code to be run before the application. It just wasn't implemented on Slurm.

I can do it

**Comment by raimis** [comment]
This is partially fixed by #540.

**Comment by j3mdamas** [comment]
OK, so here's what I'm going to do:

- [ ] I've explained @raimis how to use the config file so that the code added to the cli.py in https://github.com/Acellera/htmd/pull/540/files is not needed. I will revert that code (for 1.12.0).
- [ ] Right now, the `environment` variable of SlurmQueue and LsfQueue behaves differently (mainly due to have been developed separately). As such, I will deprecate it and create two new variables: `envvars` (to pass environment variables from submission node to running node) and `prerun` (to run arbitrary lines of code like module loading before running the job). This is for 1.12.0.
- [ ] I will create a new issue for 1.14.x where I'll create a `_BatchQueue` helper class to father Slurm, LSF and PBS, and uniformize their variables through properties.

João
